modules/azure/src/main/java/org/testcontainers/azure/AzureEventhubsEmulatorContainer.java

```diff
@@ -0,0 +1,106 @@
+package org.testcontainers.azure;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+/**
+ * Testcontainers implementation for Azure Eventhubs Emulator.
+ * <p>
+ * Supported image: {@code "mcr.microsoft.com/azure-messaging/eventhubs-emulator"}
+ * <p>
+ * Exposed ports:
+ * <ul>
+ *     <li>AMQP: 5672</li>
+ *     <li>Kafka: 9092</li>
+ * </ul>
+ */
+public class AzureEventhubsEmulatorContainer extends GenericContainer<AzureEventhubsEmulatorContainer> {
+
+    private static final int DEFAULT_AMQP_PORT = 5672;
+
+    private static final int DEFAULT_KAFKA_PORT = 9092;
+
+    private static final String CONNECTION_STRING_FORMAT =
+        "Endpoint=sb://%s:%d;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true;";
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse(
+        "mcr.microsoft.com/azure-messaging/eventhubs-emulator"
+    );
+
+    private AzuriteContainer azuriteContainer;
+
+    private MountableFile config;
+
+    /**
+     * @param dockerImageName specified docker image name to run
+     */
+    public AzureEventhubsEmulatorContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+
+        waitingFor(Wait.forLogMessage(".*Emulator Service is Successfully Up!.*", 1));
+        withExposedPorts(DEFAULT_AMQP_PORT, DEFAULT_KAFKA_PORT);
+    }
+
+    @Override
+    public void start() {
+        if (this.azuriteContainer == null) {
+            this.azuriteContainer =
+                new AzuriteContainer(AzuriteContainer.DEFAULT_IMAGE_NAME.withTag("3.33.0")).withNetwork(getNetwork());
+        }
+        this.azuriteContainer.start();
+
+        super.start();
+    }
+
+    @Override
+    public void stop() {
+        super.stop();
+        if (this.azuriteContainer != null) {
+            this.azuriteContainer.stop();
+        }
+    }
+
+    /**
+     * Provide the broker configuration to the container.
+     *
+     * @param config The file containing the broker configuration
+     * @return this
+     */
+    public AzureEventhubsEmulatorContainer withConfig(final MountableFile config) {
```

Comment:
```suggestion
    public AzureEventhubsEmulatorContainer withConfig(final Transferable config) {
```

Reply:
Should I update the Azurite config too similarly? It feels wrong to use two different approaches...

Reply:
Using transferable for the config file makes sense because the json file can be built programmatically. I do not expect the same for the certificates. WDYT?

Reply:
To be honest I don't see the big difference but this might be just because I am not that familiar with these two types and the exact intent behind them (hopefully I will learn now). 
We can generate (or download from a secure place) a Java `KeyStore` without ever writing it to file, also if you want to generate a configuration json, it is a small step to write it to a temp file and then simply create a `MountableFile` with it. Or is the difference more nuanced?

Reply:
`MountableFile` allows to copy files from host. Meanwhile `Transferable`, the interface it allows to pass a string that can be generated programmatically. Besides, it also accepts `MountableFile` because it implements `Transferable`.

Using `Transferable`, the API would be flexible enough for the user to choose.

Reply:
Okay, thank you for the explanation!